scrapy/downloadermiddlewares/httpproxy.py

```diff
@@ -26,7 +26,7 @@ def _get_proxy(self, url, orig_type):
         proxy_url = urlunparse((proxy_type or orig_type, hostport, '', '', '', ''))
 
         if user:
-            user_pass = '%s:%s' % (unquote(user), unquote(password))
+            user_pass = to_bytes('%s:%s' % (unquote(user), unquote(password)))
```

Comment:
When nothing works, we should stick to ascii and provide a way to override via settings. I think is better than guessing what it the most common, because most common is not all the cases.
Makes sense?